scripts/generate_pkcs7_tests.py

```diff
@@ -37,7 +39,15 @@ class TestData:
     Take in test_suite_pkcs7.data file.
     Allow for new tests to be added.
     """
-    mandatory_dep = "MBEDTLS_MD_CAN_SHA256"
+    # In Mbed TLS 3.6, PKCS7 code belongs to the legacy domain
+    # (see transition-guards.md for more information) thus use
+    # MBEDTLS_MD_CAN_SHA256 dependency symbol instead of PSA_WANT_ALG_SHA_256.
+    mandatory_dep = "MBEDTLS_MD_CAN_SHA256" if is_mbedtls_3_6() else "PSA_WANT_ALG_SHA_256"
+
+    # OR helper function but it does not help with the difficulty to decide
+    # if some legacy domain code is involved or not.
+#   mandatory_dep = test_case.legacy_domain_hash_dependency_symbol("PSA_ALG_SHA_256")
```

Comment:
This feels like it's an extension of `psa_information.psa_want_symbol`. Maybe even an optional argument
`psa_information.psa_want_symbol('PSA_ALG_SHA_256', psa_information.MBEDTLS36.USE_PSA_CRYPTO)`?

Reply:
The optional argument would be the domain of the code for which we want to express the dependency on a given hash, right?

Reply:
Yes.

Though on second thoughts the function should not be called `psa_want_symbol`, because that name already implies a PSA dependency. A new function in `test_case.py` does sound better, with a domain argument.

Reply:
As I said on Slack, I think for hashes we can distinguish the domain with:
- things in the PSA or pure-TLS-1.3 domain directly use `PSA_WANT` or call `psa_want_symbol`;
- things in the legacy or USE domain use a new helper function.
(Which is the current state of the PR.)

However that only works because hashes use the same macro for the legacy and USE domains. For other things we're not going to be that lucky. We could have  two functions there, but then things wouldn't be uniform.

So, I think I have a preference for a function with a domain argument. Then we don't even need to separate by type of mechanisms (hashes vs cipher vs etc.)

For example:
```py
def psa_or_36_feature_macro(psa_alg, domain36):
    if domain36 = DOMAIN36_PSA or not build_tree.is_mbedtls_3_6():
        return psa_information.psa_want_symbol(psa_alg)

    if psa_alg in hashes_for_36:
        return hashes_for_36[psa_alg]

    if domain36 = DOMAIN36_LEGACY and psa_alg in foo_for_36_legacy:
        return foo_for_36_legacy[psa_alg]
    if domain36 = DOMAIN36_USE_PSA and psa_alg in foo_for_36_use_psa:
        return foo_for_36_use_psa[psa_alg]
    # etc for each family of algs

    raise ValueError
```
(Or we could have a single dict mapping `(psa_alg, domain36)` to 3.6 feature macro (which would then need to have duplicated entries for hashes) or any other implementation strategy.)

What I mean is we can have a single function that acts as a drop-in replacement for `psa_want_symbol()` except it has an extra argument for the 3.6 domain, and I think this has my preference at the moment.